File: mac/Sources/FocusKit/Models/TimerMode.swift
import Foundation

public enum TimerMode: String, CaseIterable, Codable, Identifiable {
    case focus
    case shortBreak
    case longBreak

    public var id: String { rawValue }

    public var label: String {
        switch self {
        case .focus: return "Focus"
        case .shortBreak: return "Short Break"
        case .longBreak: return "Long Break"
        }
    }

    public var defaultMinutes: Int {
        switch self {
        case .focus: return 60
        case .shortBreak: return 5
        case .longBreak: return 15
        }
    }

    /// Only focus time is tracked; breaks are not work.
    public var isTracked: Bool { self == .focus }
}

/// Focus rounds completed before a long break is suggested.
let roundsBeforeLongBreak = 4

File: windows/Focus.Core/Models/TimerMode.cs
namespace Focus.Core;

public enum TimerMode { Focus, ShortBreak, LongBreak }

public static class TimerModes
{
    /// Focus rounds completed before a long break is suggested.
    public const int RoundsBeforeLongBreak = 4;

    public static readonly TimerMode[] All = { TimerMode.Focus, TimerMode.ShortBreak, TimerMode.LongBreak };

    /// The wire name, shared with the Mac app's preference keys.
    public static string Key(this TimerMode mode) => mode switch
    {
        TimerMode.Focus => "focus",
        TimerMode.ShortBreak => "shortBreak",
        _ => "longBreak"
    };

    public static string Label(this TimerMode mode) => mode switch
    {
        TimerMode.Focus => "Focus",
        TimerMode.ShortBreak => "Short Break",
        _ => "Long Break"
    };

    public static int DefaultMinutes(this TimerMode mode) => mode switch
    {
        TimerMode.Focus => 60,
        TimerMode.ShortBreak => 5,
        _ => 15
    };

    /// Only focus time is tracked; breaks are not work.
    public static bool IsTracked(this TimerMode mode) => mode == TimerMode.Focus;
}
